### PR TITLE
Plans details table

### PR DESCRIPTION
**Pull request by gildub**
Resolves https://github.com/konveyor/virt-ui/issues/57

### CODE REVIEW

**Comment by mturley** [comment]
@gildub I noticed your status on slack mentioned needing to add stuff to this page, is this PR ready for review or should it be a draft?

**Comment by gildub** [comment]
@mturley, I've updated it. Thanks

File: src/app/Plans/components/VMMigrationDetails.tsx
Comment (mturley) [inline src/app/Plans/components/VMMigrationDetails.tsx:80]: Can we move these to their own mocks file in `@app/queries/mocks/migrations.mock.ts`, or maybe add them to `plans.types.ts`? Even if we're going to replace them later it would de-clutter this a bit.

File: src/app/Plans/components/VMMigrationDetails.tsx
Comment (mturley) [inline src/app/Plans/components/VMMigrationDetails.tsx:131]: Just in case, let's use the `isItemSelected` function returned from `useSelectionState` to determine expanded state instead of `.includes()`, i'm worried we'll lose this state when queries refetch because of referential equality with stale data.

File: src/app/Plans/components/VMMigrationDetails.tsx
Comment (mturley) [inline src/app/Plans/components/VMMigrationDetails.tsx:141]: Can you remove the commented line here?

File: src/app/Plans/components/VMMigrationDetails.tsx
Comment (mturley) [inline src/app/Plans/components/VMMigrationDetails.tsx:193]: I think we can remove the `variant` prop here, this table doesn't look compact in the mocks. The VMStatusTable should stay compact though.

File: src/app/queries/mocks/plans.mock.ts
Comment (mturley) [inline src/app/queries/mocks/plans.mock.ts:162]: Ah, I hadn't realized we already had `IMigration`. What's the reason for having both this and the `ITempMockMigration` from above?

**Review by mturley — CHANGES_REQUESTED** [review]
This is looking really good, I just have a few minor comments.

File: src/app/queries/mocks/plans.mock.ts
Comment (gildub) [reply src/app/queries/mocks/plans.mock.ts:240]: @mturley, this is temporary to avoid touching back and forth the structure until we have more visibility on those structure.
But that's annoying. Let me see if I can do better now that I have the UI in place.

File: src/app/Plans/components/VMMigrationDetails.tsx
Comment (gildub) [reply src/app/Plans/components/VMMigrationDetails.tsx]: Yes, sorry about that, I consolidated the migration mock data.

File: src/app/queries/mocks/plans.mock.ts
Comment (gildub) [reply src/app/queries/mocks/plans.mock.ts]: I've removed it after consolidating the mock migrations.

File: src/app/Plans/components/VMMigrationDetails.tsx
Comment (gildub) [reply src/app/Plans/components/VMMigrationDetails.tsx:86]: Done.

File: src/app/Plans/components/VMMigrationDetails.tsx
Comment (mturley) [reply src/app/Plans/components/VMMigrationDetails.tsx]: No need to apologize! 👍

**Comment by gildub** [comment]
@vconzola, please review.

**Review by mturley — DISMISSED** [review]
LGTM, but I'll wait for @vconzola's review before merging.

**Comment by gildub** [comment]
As discussed today during Scrum, all actions but the enable edit/remove options from Dropdown (Kebab) for the `PlansTable`.

I suggest to do that last one into a separate PR otherwise I'll come back to it first thing tomorrow.

**Comment by gildub** [comment]
@vconzola, please check it out.

**Comment by mturley** [comment]
yeah, I think that can be considered out of scope since it's on the main plans page and not the plan details page. We can open an issue for that.

**Review by mturley — APPROVED** [review]
Approved by Vince over Slack. Opened #98 to follow up about the kebab actions.

Nice work @gildub !

**Pull request merged**
